### PR TITLE
Add Initial Support For EKS Fargate Testing In Terraform

### PR DESCRIPTION
**Pull request by sethAmazon**

### CODE REVIEW

File: terraform/validation/main.tf [hunk 1]
```diff
@@ -49,6 +49,7 @@ data "template_file" "docker_compose" {
     incoming_packets_alarm = var.incoming_packets_alarm
 
     cortex_instance_endpoint = var.cortex_instance_endpoint
+    rollup                   = var.rollup
```
Comment (khanhntd) [inline terraform/validation/main.tf:52]: What does roll-up for? Any documents on that since I could not fine one.

Comment (rltoSD) [reply terraform/validation/main.tf:52]: @khanhntd  Hi, rollup is a new feature for cwmetricsvalidator to enable or disable dimension rollup for expected test templates. By default it was set to true earlier, but with this newly implemented feature we can also turn it off.

File: terraform/eks/main.tf [hunk 2]
```diff
@@ -152,7 +167,7 @@ module "validator" {
   region                       = var.region
   testing_id                   = module.common.testing_id
   metric_namespace             = "${module.common.otel_service_namespace}/${module.common.otel_service_name}"
-  sample_app_endpoint          = length(kubernetes_service.sample_app_service) > 0 ? "http://${kubernetes_service.sample_app_service.0.load_balancer_ingress.0.hostname}:${module.common.sample_app_lb_port}" : ""
+  sample_app_endpoint          = length(kubernetes_ingress.app) > 0 && var.deployment_type == "fargate" ? "http://${kubernetes_ingress.app.0.load_balancer_ingress.0.hostname}:80" : length(kubernetes_service.sample_app_service) > 0 ? "http://${kubernetes_service.sample_app_service.0.load_balancer_ingress.0.hostname}:${module.common.sample_app_lb_port}" : ""
```
Comment (khanhntd) [inline terraform/eks/main.tf]: For this port, can we set as a variable port instead of default one?

File: terraform/eks/main.tf [hunk 3]
```diff
@@ -108,6 +111,18 @@ resource "kubernetes_service_account" "aoc-role" {
   automount_service_account_token = true
 }
 
+resource "kubernetes_service_account" "aoc-fargate-role" {
+  metadata {
+    name      = "aoc-fargate-role-${module.common.testing_id}"
+    namespace = "default"
+    annotations = {
+      "eks.amazonaws.com/role-arn" : "arn:aws:iam::611364707713:role/ServiceAccount-eks-test-aoc-role"
```
Comment (khanhntd) [inline terraform/eks/main.tf]: Can we set the account Id to a dynamic one instead of default one?

File: .gitignore [hunk 4]
```diff
@@ -58,3 +58,5 @@ override.tf.json
 terraform.rc
 
 !gradle/wrapper/gradle-wrapper.jar
+
+*/.DS_Store
```
Comment (jefchien) [inline .gitignore]:
```suggestion
.DS_Store
**/.DS_Store
```
https://git-scm.com/docs/gitignore
>A leading "**" followed by a slash means match in all directories. For example, "**/foo" matches file or directory "foo" anywhere, the same as pattern "foo".

File: docs/game.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+---
```
Comment (jefchien) [inline docs/game.yml]: I don't think we need to duplicate this file in our repo. We can probably just link to the example in our docs. https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/docs/examples/2048/2048_full.yaml

File: terraform/eks/main.tf (hunk 3 above)
Comment (sethAmazon) [reply terraform/eks/main.tf]: This I was going to change in the next phase.

Comment (khanhntd) [reply terraform/eks/main.tf]: Got cha. Should  catch up with the plan for the current phase.

File: terraform/eks/container-insights-agent/aoc_fargate_deploy.yml [hunk 6]
```diff
@@ -0,0 +1,557 @@
...
+kind: ConfigMap
+metadata:
+  name: adot-collector-config
+  namespace: default
+  labels:
+    app: aws-adot
+    component: adot-collector-config
+data:
+  adot-collector-config: |
+    receivers:
```
Comment (jefchien) [inline terraform/eks/container-insights-agent/aoc_fargate_deploy.yml]: Are we using a fixed config for each of the tests? Can we use a template and render setup?

File: terraform/eks/container-insights-agent/config_map_fargate.yml [hunk 7]
```diff
@@ -0,0 +1,442 @@
...
+kind: ConfigMap
+metadata:
+  name: adot-collector-config
+  namespace: default
+  labels:
+    app: aws-adot
+    component: adot-collector-config
+data:
+  adot-collector-config: |
+    receivers:
```
Comment (jefchien) [inline terraform/eks/container-insights-agent/config_map_fargate.yml:11]: Is this the same config as in `aoc_fargate_deploy.yml`?

File: terraform/validation/main.tf (hunk 1 above)
Comment (khanhntd) [reply terraform/validation/main.tf:52]: Reasonable. Thanks for explaining.

**Review by khanhntd — APPROVED** [review]
LGTM.

File: terraform/eks/main.tf (hunk 2 above)
Comment (jefchien) [reply terraform/eks/main.tf]: It's a bit difficult to parse nested ternary statements. Can we use parentheses to break it up?

File: docs/game.yml (hunk 5 above)
Comment (sethAmazon) [reply docs/game.yml]: That link does not use the default namespace. It uses another namespace.

File: terraform/eks/container-insights-agent/aoc_fargate_deploy.yml (hunk 6 above)
Comment (sethAmazon) [reply terraform/eks/container-insights-agent/aoc_fargate_deploy.yml]: I can add that in later commits. So we can get the testing in.

File: docs/game.yml (hunk 5 above)
Comment (jefchien) [reply docs/game.yml]: Is that the only difference?

File: terraform/eks/container-insights-agent/config_map_fargate.yml (hunk 7 above)
Comment (sethAmazon) [reply terraform/eks/container-insights-agent/config_map_fargate.yml:11]: Yes the first file contains the fill config. I can delete it. Sorry I was using it as a reference.

File: docs/game.yml (hunk 5 above)
Comment (sethAmazon) [reply docs/game.yml]: yes

Comment (sethAmazon) [reply docs/game.yml]: Eventually this will be removed when I get the cluster creation fully working.

Comment (jefchien) [reply docs/game.yml]: So, could you explain that in the doc instead of adding the file? Like
> * Create Ingress controller fargate
Download config [here](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/docs/examples/2048/2048_full.yaml) and replace all instances of namespace `game-2048` with `default`.

**Review by jefchien — APPROVED** [review]

File: terraform/eks/container-insights-agent/logs_sample_fargate.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      app: flblogproducer2eks
+  template:
+    metadata:
+      labels:
+        app: flblogproducer2eks
+    spec:
+      containers:
+        - name: flblogproducer2eks
+          #image: 803860917211.dkr.ecr.us-west-2.amazonaws.com/alpine-log-producer:latest
+          image: 803860917211.dkr.ecr.us-west-2.amazonaws.com/python-logger:300sec
```
Comment (hossain-rayhan) [inline terraform/eks/container-insights-agent/logs_sample_fargate.yml:21]: This image is in a personal ECR account. Let's replace it with `nginx:latest` and remove the command section.

File: terraform/eks/container-insights-agent/logs_sample_fargate.yml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: flblogproducer2eks
```
Comment (hossain-rayhan) [inline terraform/eks/container-insights-agent/logs_sample_fargate.yml:4]: Let's rename this to `app-container` in 5 places. This is not related to Fluent Bit. It was added only for testing purposes.
